Sources/FluentKit/Properties/Parent.swift

```diff
@@ -1,207 +1,231 @@
 @propertyWrapper
-public final class Parent<To>: AnyField, AnyEagerLoadable
-    where To: Model
-{
-    // MARK: ID
+public final class Parent<T>: AnyField, AnyEagerLoadable where T: ModelIdentifiable {
+    public typealias To = T
+    typealias Implementation = AnyProperty & AnyEagerLoadable
 
-    public typealias Value = To.IDValue
+    @Field public var id: To.IDValue
 
-    private var cachedLabel: String?
     private var eagerLoadedValue: To?
+    private var implemntation: ParentImplementation!
 
-    @Field
-    public var id: To.IDValue
-
-    // MARK: Wrapper
+    var key: String { return self.$id.key }
+    var inputValue: DatabaseQuery.Value? {
+        get { self.$id.inputValue }
+        set { self.$id.inputValue = newValue }
+    }
 
     public var wrappedValue: To {
-        get {
-            guard let eagerLoaded = self.eagerLoadedValue else {
-                fatalError("Parent relation not eager loaded, use $ prefix to access")
-            }
-            return eagerLoaded
+        guard let value = self.eagerLoadedValue else {
+            if _isOptional(To.self) { return Void?.none as! To }
```

Comment:
What does this line do? `fatalError` seems like the appropriate thing to do whether or not the relation is optional. Returning `nil` if we haven't done a prefetch would be incorrect if there was actually a value in the DB.